### ISSUE TITLE
Replace cryptography with secure & audited packages

### ISSUE THREAD

**Issue by paulmillr** [ConsenSys-archive/ganache#3426]
Metamask is planning to update cryptography to new, audited libraries aka `noble-crypto`. Ganache (metamask dep) still uses `secp256k1` and a few other projects, which haven't passed a security audit. It would be great to have all of them updated, to make both projects use same libs.

In January, EF-funded https://github.com/ethereum/js-ethereum-cryptography v1.0 was published. It contains audited cryptography needed for ETH.

https://github.com/MetaMask/metamask-extension/issues/14928#issuecomment-1194515973

**Comment by davidmurdoch** [comment]
Interesting; `noble-crypto` is likely a few orders of magnitude slower than `secp256k1`. `secp256k1` uses bitcoin-core's own native secp256k1 implementation; my intuition tells me this isn't likely to have been compromised.

We _do_ fall back to a pure JavaScript implementation of secp256k1 (via the elliptic package, which I noticed you've analyzed as well) if the native version can't be loaded. Switching out our JS fallback to use noble's implementation is probably not a problem and would likely improve our pure JS performance as well as reduce our bundle size.

For the _native_ secp256k1 we'll need to measure performance to ensure switching won't cause regressions.

**Comment by paulmillr** [comment]
Yeah, if you can use native ones, it's great to keep it around. I'm only asking to replace the js fallback with noble.

For benchmarks, see [readme](https://github.com/paulmillr/noble-secp256k1#speed). It's faster than native openssl for signing.

**Comment by paulmillr** [comment]
I think we've even developed [the file](https://github.com/ethereum/js-ethereum-cryptography/blob/master/src/secp256k1-compat.ts) to make switches like this one easy.
